File: calculadora.py
# -*- coding: utf-8 -*-
"""
Created on Sat Dec  4 12:02:08 2021

@author: SERGIO
2do Examen INF - 319
Pregunta 1
"""
#Suma
def Suma(x, y):
    return x + y

# Resta
def Resta(x, y):
    return x-y
# Multiplica
def Multiplicar(x, y):
    return x * y

# División
def Dividdir(x, y):
    if(y==0):
        return "no hay division entre cero"
    else:
       return int(x / y) 

#calculadora
def calculadora(entrada,num1,num2):
    if entrada == '1':
        print("--La suma es: ",Suma(num1,num2))
    elif entrada == '2':
        print("--La Resta es: ",Resta(num1,num2))
    elif entrada == '3':
        print("--La Multiplicacion es: ",Multiplicar(num1,num2))
    elif entrada == '4':
        print("--La division es: ", Dividdir(num1,num2))
    else:
        print("ERROR numero incorrecto, intente de nuevo")

#Menu
while(True):
    print("******CALCULADORA EN PYTHON CON FUNCIONES******")
    print("1.Suma")
    print("2.Resta")
    print("3.Multiplicar")
    print("4.Dividir")
    # Operación a realizar
    entrada = input("Que operacion desea realizar...digite (1/2/3/4): ")
    n1 = int(input("Primer número: "))
    n2 = int(input("segundo número: "))
    #Llamada a la funcion calculadora
    calculadora(entrada,n1,n2)

File: calculadora.scala
//Calculadora con funciones en Scala
//Sergio Luis Huanca Cuellar
//Segundo Parcial - problema 1

object calculadora {
  //funcion sumar
  def sumar(x: Int, y: Int): Int =  x + y
  //funcion restar
  def restar(x: Int, y: Int): Int = x - y
  
  def multiplicar(x: Int, y: Int): Int =   x * y

  def dividir(x: Int, y: Int): Int = 
    y match {
      case 0 => -1
      case _ => x/y
  }
//funcion calculadora
 def calcu(x:Int, y:Int, operacion:String):Int={
  operacion match {
  case "suma"=>sumar(x,y)
  case "resta"=>restar(x,y)
  case "multi"=>multiplicar(x,y)
  case "divi"=>dividir(x,y)
  case _ => -1
  }
} 

def main(args: Array[String]):Unit = {
  println("*****CALCULADORA EN SCALA CON FUNCIONES******")
  //valores
  val a = 100
  val b = 20
  println("Los valores a operar son: "+ a +","+ b)
//suma
  println("Suma: "+calcu(a,b,"suma")) // numero a operar  4 5 
//resta
  println("Resta: "+ calcu(a,b,"resta"))
  //multi
  println("Multiplicacion: "+ calcu(a,b,"multi"))
  //divi
  println("Division: "+ calcu(a,b,"divi"))
}
}